<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Prediction intervals and polygenic scores</title>
    <meta charset="utf-8" />
    <meta name="author" content="Fred Boehm" />
    <meta name="date" content="2021-11-05" />
    <script src="2021-november-05-zhou-lab-meeting-slides_files/header-attrs/header-attrs.js"></script>
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Prediction intervals and polygenic scores
### Fred Boehm
### November 5, 2021

---






## Two recent articles address prediction intervals in polygenic risk scores

1. Translating polygenic risk scores for clinical use by estimating the confidence bounds of risk prediction by Sun, et al. (Nature communications, 2021)

2. Large uncertainty in individual PRS estimation impacts PRS-based risk stratification by Ding, et al. (Biorxiv, 2021)



---

## Overview of @sun2021translating

&lt;img src="figures/sun-et-al/41467_2021_25014_Fig1_HTML.png" width="800" height="500" /&gt;

---

class: inverse center middle

# Methods


---

## Mondrian Cross-Conformal Prediction (MCCP)

- implementation of conformal prediction in classification  
- guarantees validity of the predictor for both cases and controls  
- used to estimate confidence levels of risk prediction in a sample where disease status is known


---

## Procedure

1. Compute PRS for each subject    
1. Divided subjects into training and test sets  
1. Partition training set into `\(n\)` equal subsets  
1. Retain one of the `\(n\)` subsets as the "calibration set" to calculate the MCCP probability values (Eq 1) 
1. Remaining `\(n-1\)` subsets are the "proper training set"  
1. Fit logistic regression models on the proper training set  

$$ p_y^i = \frac{| \lbrace j: y_i = y, NCM_j \ge NCM_i\rbrace |}{ N\_cal_y + 1: y_i = y}$$
- `\(i\)` denotes subject index and `\(y\)` denotes class - case or control  
- `\(N\_cal_y\)` denotes number of cases (or controls) in calibration set  
- Thus, every test subject is assigned a probability of being a case and a probability of being a control  


---

## Procedure 

1. Predict on both the calibration and testing sets  
1. Calculate a nonconformity measure on every subject in the calibration and testing sets  
1. Use NCMs in calibration set assign probability values for case and control to subjects in testing set  
1. Repeat procedure `\(n\)` times, using each of the `\(n\)` subsets exactly once as calibration set  
1. Average the `\(n\)` probability values for each test subject  

---

## Simulations: Design

- 10,000 subjects from iPSYCH study  
- used real genotypes on Chr 2  
- simulated case and control phenotype values 
- varied polygenicity: 0.001, 0.01, 0.1, 0.2  
- varied heritability: 0.3, 0.5, 0.8  
- varied case prevalence: 0.01, 0.05, 0.1, 0.2  
- discovery set proportion varied: 0.2, 0.5, and 0.7  
- PRS calculated with thresholding (p &lt; 0.05) and pruning (LD `\(r^2 &lt; 0.1\)`)  

---

## Simulations: PRS construction  

- weighted sum of effects for pruned SNPs with MAF `\(\ge 0.01\)`
- effect sizes taken from discovery GWAS or reference GWAS  

---

## Prediction outcomes 

- MCCP produces a prediction set - containing 0, 1, or 2 labels - for every test subject  
- Differs from classification approaches that produce exactly one label for every test subject  

---

## Calibration assessment  

- Reliability curves to assess calibrations by MCCP and LR  
- Divide data into five folds  
- Build MCCP and LR on four folds and make predictions on fifth fold  
- Observed error is proportion of incorrect predictions against true case-control status

`$$Observed_{err}(LR) = \sum \lbrace p^i &gt; 1 - \alpha: y = 0 || p^i &lt; \alpha: y = 1\rbrace$$`

`$$Observed_{err}(MCCP) = \sum \lbrace p_y^i \le \alpha : y \rbrace$$` 

---

## Evaluation metrics 

MCCP performance measured by *validity* and *coverage*

- A prediction is valid if error frequency is less than or equal to chosen error rate `\(\alpha\)`. 

- Coverage is defined as percentage predicted as case or control  

---

class: inverse center middle

# Results 


---

## Figure 2, Simulations &amp; Calibration plots 

&lt;img src="figures/sun-et-al/41467_2021_25014_Fig2_HTML.png" width="300" height="500" /&gt;

.pull-right[
1.  Logistic regression underestimates error rate for small error rates &amp; overestimates error rate for large error rates  
1. MCCP has perfectly calibrated error rates across all considered scenarios  
]


---

## MCCP vs. Empirical method in real data analysis

- Empirical method:  
    - Top `\(x\)` proportion ( `\(x &lt; 0.5\)` ) are one class  
    - Bottom `\(x\)` proportion ( `\(x &lt; 0.5\)` ) are second class  
- Coverage for "empirical method" `\(= 2x\)`  
- Empirical method doesn't offer individual-level prediction intervals  

---

## Figure 3, Performance with real data (PRS only)  

&lt;img src="figures/sun-et-al/41467_2021_25014_Fig3_HTML.png" width="800" height="500" /&gt;

---

## Figure 4, Performance with real data (PRS and covariates)

&lt;img src="figures/sun-et-al/41467_2021_25014_Fig4_HTML.png" width="800" height="400" /&gt;

---

class: inverse center middle


# Discussion 


---

## Discussion Highlights  

- MCCP risk estimates have the "validity" property: 
    - in the long run, prediction errors fall below a preset error rate  
- MCCP may aid clinicians by providing individual-level predictions  
- Future work may apply MCCP to settings where genetic and other subject-level data are available: 
    - lifestyle, SES, environmental exposures  
- Other machine learning approaches - knn, random forest, svm, deep nn - may be combined with MCCP in risk prediction  

---

## Implications for our PRS methods research  

- DBSLMM and MCCP to get prediction intervals for quantitative traits    
    - Extend MCCP from binary classification to regression setting  



---

class: center, middle

# Thanks!

Slides created via the R packages:

[**xaringan**](https://github.com/yihui/xaringan)&lt;br&gt;
[gadenbuie/xaringanthemer](https://github.com/gadenbuie/xaringanthemer)

The chakra comes from [remark.js](https://remarkjs.com), [**knitr**](http://yihui.name/knitr), and [R Markdown](https://rmarkdown.rstudio.com).
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"slideNumberFormat": "%current%",
"highlightStyle": "github",
"highlightLines": true,
"ratio": "16:9",
"countIncrementalSlides": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
