---
poster_width: "36in"
poster_height: "48in"
main_topsize: 0.25 #percent coverage of the poster
main_bottomsize: 0.1
#ESSENTIALS
title: '**Statistical power to detect QTL improves after inferring allelic series**'
title_textsize: "60pt"
author:
  - name: '**Frederick J. Boehm**'  
    affil: 1
    main: true
    orcid: '0000-0002-1644-5931'
    email: frederick.boehm@gmail.com
    website: https://fboehm.us
  - name: Clare M. Smith
    affil: 2
  - name: Richard Baker
    affil: 1
  - name: Megan Proulx
    affil: 1
  - name: Daniel M. Gatti
    affil: 3
  - name: Michael Kiritsy
    affil: 1
  - name: Sherry Kurtz
    affil: 4
  - name: Karen Elkins
    affil: 4
  - name: Gillian Beamer
    affil: 5
  - name: Christopher M. Sassetti
    affil: 1
affiliation:
  - num: 1
    address: University of Massachusetts Medical School
  - num: 2
    address: Duke University
  - num: 3
    address: College of the Atlantic
  - num: 4
    address: U.S. Food and Drug Administration
  - num: 5
    address: Tufts University
main_findings:
  - "**Enhance QTL detection** with less flexible models **informed by allelic series**."
logoleft_name: https&#58;//raw.githubusercontent.com/brentthorne/posterdown/master/images/betterhexlogo.png
logoright_name: https&#58;//raw.githubusercontent.com/brentthorne/posterdown/master/images/betterhexlogo.png
logocenter_name: qr-fboehmus.png
output: 
  posterdown::posterdown_betterport:
    self_contained: false
    pandoc_args: --mathjax
    number_sections: false
bibliography: ../research.bib
link-citations: true
---

```{r, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      tidy = FALSE,
                      message = FALSE,
                      fig.align = 'center',
                      out.width = "100%")
options(knitr.table.format = "html") 
```

# Introduction

```{r vitruvian}
knitr::include_graphics("../figs/vitruvian-mouse-eleni.jpg")
```

Systems genetics studies in model organisms have informed our understanding of biology and disease processes. 
Multiparental crosses of more than two founders incorporate greater levels of genetic diversity than two-parent crosses.
Two widely used mouse resources are the Collaborative Cross and Diversity Outbred mice. 
Collaborative Cross includes approximately 60 inbred mouse lines, each of which incorporates DNA from eight founder lines.
Diversity Outbred mice are outbred, highly heterozygous, and are genetic mosaics of the same eight founders.
Quantitative trait locus (QTL) mapping studies have identified genetic loci affecting diverse traits, including phenotypes related to immune responses to *M. tuberculosis* infection, in Diversity Outbred and Collaborative Cross mice. 



```{r, include=FALSE}
knitr::write_bib(c('posterdown', 'rmarkdown','pagedown'), 'packages.bib')
```

```{r mpp, eval = TRUE, echo = FALSE, out.height=400, out.width="65%"}
##############################
# HS (light background)
# from kbroman/QTLfigs on github
##############################

n.lines <- 80

bgcolor <- "white"
color <- qtl2::CCcolors

#pdf("Figs/hs_light.pdf", width=9.75, height=6.5, pointsize=16, onefile=TRUE)
par(mar=rep(0.1,4),las=1,fg="black",col="black",col.axis="black",col.lab="black",
    bg=bgcolor, bty="n")
plot(0,0,xlim=c(0,864),ylim=c(25,480),xaxt="n",yaxt="n",xlab="",ylab="",type="n")

u <- par("usr")
text(u[1],460,expression(G[0]),adj=c(0,0.5))

xloc <- 10
yloc <- 480
for(i in 1:8) {
  rect(xloc,yloc,xloc+8,yloc-40,col=color[i],border=color[i], lend=1, ljoin=1)
  rect(xloc+20,yloc,xloc+28,yloc-40,col=color[i],border=color[i], lend=1, ljoin=1)
  text(xloc+14,yloc-50,LETTERS[i],adj=c(0.5,1))

  xloc <- xloc+38+35+46
}

set.seed(11716365)
source(here::here("R/meiosis_func.R"))
gen <- vector("list", n.lines)
for(i in 1:n.lines)
  gen[[i]] <- create.par(100, sample(1:8, 2))
wh <- seq(1, n.lines, by=5)

xloc <- 10
yloc <- yloc - 80
for(i in 1:15) {
  g <- c(gen[[wh[i]]][[1]][2,1], gen[[wh[i]]][[2]][2,1])
  rect(xloc,yloc,xloc+8,yloc-40,col=color[g[1]],border=color[g[1]], lend=1, ljoin=1)
  rect(xloc+20,yloc,xloc+28,yloc-40,col=color[g[2]],border=color[g[2]], lend=1, ljoin=1)

  xloc <- xloc+(38+35+46)/2
}
text(u[1],yloc-20,expression(G[1]),adj=c(0,0.5))

mysample <- function(n)
  as.numeric(t(apply(matrix(1:n, ncol=2, byrow=TRUE), 2, sample)))

old <- gen[mysample(n.lines)]
for(i in 1:(n.lines/2)) {
  gen[[i*2-1]] <- cross(old[[i*2-1]], old[[i*2]])
  gen[[i*2]] <- cross(old[[i*2-1]], old[[i*2]])
}

xloc <- 10
yloc <- yloc - 80
for(i in 1:15) {
  thegen <- gen[[wh[i]]]
  g <- c(thegen[[1]][2,1], thegen[[2]][2,1])

  rect(xloc,yloc,xloc+8,yloc-40,col=color[g[1]],border=color[g[1]], lend=1, ljoin=1)
  rect(xloc+20,yloc,xloc+28,yloc-40,col=color[g[2]],border=color[g[2]], lend=1, ljoin=1)

  mult <- 40/thegen[[1]][1,ncol(thegen[[1]])]
  xxloc <- c(xloc, xloc+20)
  for(k in 1:2) {
    temp <- thegen[[k]]
    for(j in 2:ncol(temp)) {
      rect(xxloc[k],yloc-40+temp[1,j]*mult,xxloc[k]+8,yloc-40+temp[1,j-1]*mult,
           col=color[temp[2,j]],border=color[temp[2,j]], lend=1, ljoin=1)
    }
  }

  xloc <- xloc+(38+35+46)/2
}
text(u[1],yloc-20,expression(G[2]),adj=c(0,0.5))


for(kk in 3:10) {
  old <- gen[mysample(n.lines)]
  for(i in 1:(n.lines/2)) {
    gen[[i*2-1]] <- cross(old[[i*2-1]], old[[i*2]])
    gen[[i*2]] <- cross(old[[i*2-1]], old[[i*2]])
  }
}

xloc <- 10
yloc <- yloc - 80
for(i in 1:15) {
  thegen <- gen[[wh[i]]]
  g <- c(thegen[[1]][2,1], thegen[[2]][2,1])

  rect(xloc,yloc,xloc+8,yloc-40,col=color[g[1]],border=color[g[1]], lend=1, ljoin=1)
  rect(xloc+20,yloc,xloc+28,yloc-40,col=color[g[2]],border=color[g[2]], lend=1, ljoin=1)

  mult <- 40/thegen[[1]][1,ncol(thegen[[1]])]
  xxloc <- c(xloc, xloc+20)
  for(k in 1:2) {
    temp <- thegen[[k]]
    for(j in 2:ncol(temp)) {
      rect(xxloc[k],yloc-40+temp[1,j]*mult,xxloc[k]+8,yloc-40+temp[1,j-1]*mult,
           col=color[temp[2,j]],border=color[temp[2,j]], lend=1, ljoin=1)
    }
  }

  xloc <- xloc+(38+35+46)/2
}
text(u[1],yloc-20,expression(G[10]),adj=c(0,0.5))


for(kk in 11:15) {
  old <- gen[mysample(n.lines)]
  for(i in 1:(n.lines/2)) {
    gen[[i*2-1]] <- cross(old[[i*2-1]], old[[i*2]])
    gen[[i*2]] <- cross(old[[i*2-1]], old[[i*2]])
  }
}

xloc <- 10
yloc <- yloc - 80
for(i in 1:15) {
  thegen <- gen[[wh[i]]]
  g <- c(thegen[[1]][2,1], thegen[[2]][2,1])

  rect(xloc,yloc,xloc+8,yloc-40,col=color[g[1]],border=color[g[1]], lend=1, ljoin=1)
  rect(xloc+20,yloc,xloc+28,yloc-40,col=color[g[2]],border=color[g[2]], lend=1, ljoin=1)

  mult <- 40/thegen[[1]][1,ncol(thegen[[1]])]
  xxloc <- c(xloc, xloc+20)
  for(k in 1:2) {
    temp <- thegen[[k]]
    for(j in 2:ncol(temp)) {
      rect(xxloc[k],yloc-40+temp[1,j]*mult,xxloc[k]+8,yloc-40+temp[1,j-1]*mult,
           col=color[temp[2,j]],border=color[temp[2,j]], lend=1, ljoin=1)
    }
  }

  xloc <- xloc+(38+35+46)/2
}
text(u[1],yloc-20,expression(G[15]),adj=c(0,0.5))

for(kk in 16:20) {
  old <- gen[mysample(n.lines)]
  for(i in 1:(n.lines/2)) {
    gen[[i*2-1]] <- cross(old[[i*2-1]], old[[i*2]])
    gen[[i*2]] <- cross(old[[i*2-1]], old[[i*2]])
  }
}

xloc <- 10
yloc <- yloc - 80
for(i in 1:15) {
  thegen <- gen[[wh[i]]]
  g <- c(thegen[[1]][2,1], thegen[[2]][2,1])

  rect(xloc,yloc,xloc+8,yloc-40,col=color[g[1]],border=color[g[1]], lend=1, ljoin=1)
  rect(xloc+20,yloc,xloc+28,yloc-40,col=color[g[2]],border=color[g[2]], lend=1, ljoin=1)

  mult <- 40/thegen[[1]][1,ncol(thegen[[1]])]
  xxloc <- c(xloc, xloc+20)
  for(k in 1:2) {
    temp <- thegen[[k]]
    for(j in 2:ncol(temp)) {
      rect(xxloc[k],yloc-40+temp[1,j]*mult,xxloc[k]+8,yloc-40+temp[1,j-1]*mult,
           col=color[temp[2,j]],border=color[temp[2,j]], lend=1, ljoin=1)
    }
  }

  xloc <- xloc+(38+35+46)/2
}
text(u[1],yloc-20,expression(G[20]),adj=c(0,0.5))

```



## Objectives

1. Traditional QTL mapping in Diversity Outbred mice [@broman2019r]
2. Infer allelic series at QTL [@crouse2018timbr]
3. QTL mapping, informed by allelic series, of traits in Collaborative Cross mice

# Methods

## Limitations of current methods

Widely used statistical models in QTL mapping in multiparental populations allow for each founder haplotype to have its own effect on trait values [@broman2019r].

$$
Y = p_A\beta_A + p_B\beta_B + p_C\beta_C + p_D\beta_D + \\
p_E\beta_E + p_F\beta_F + p_G\beta_G + p_H\beta_H + \epsilon
$$

However, many QTL have fewer than eight alleles in the Diversity Outbred and Collaborative Cross mice. Allelic series tells us the number of alleles and which founder haplotypes have which alleles. 

If a QTL in Diversity Outbred or Collaborative Cross mice has only three alleles, we may model its trait as:

$$
Y = p_1\beta_1 + p_2\beta_2 + p_3\beta_3 + \epsilon
$$

## Strategies to infer allelic series

Wesley Crouse and colleagues developed a Bayesian method called "Tree-based Inference of Multiallelism via Bayesian Regression (TIMBR)" with the goal of inferring multiallelism across the genome for a collection of genetically diverse subjects. We aim to apply its approach to infer allelic series at established QTL. 




## QTL mapping informed by allelic series

We recode founder allele probabilities according to allelic series. For example, in the case above where we have three alleles represented among the eight founders, let's suppose that the allelic series tells us that founders A, B, and C share allele 1, D and E share allele 2, and F, G, and H share allele 3. We can then relate the probabilities as:

$$
p_1 = p_A + p_B + p_C \\
p_2 = p_D + p_E \\
p_3 = p_F + p_G + p_H
$$
With the numeric values of $p_1$, $p_2$, and $p_3$, we perform standard QTL mapping across the genome. 




# Results


# References

