---
title: ""
author: "Presented by Fred Boehm"
format: 
  revealjs:
    incremental: true
    smaller: true
    scrollable: true
    theme: blood
---

## Phenotype model for admixed individuals

For individual $i = 1, \ldots, N$, SNPs $s = 1, \ldots, S$, and ancestries $k = 1, 2$, we have

$$g_{i,s,k} = x_{i,s,M}1_{y_{i,s,M} = k} + x_{i,s,P}1_{y_{i,s,P} = k}$$

- $g_{i,s,k}$ encodes allele counts that are specific to the local ancestry

## Phenotype model for admixed individuals

- Denote the causal allelic effects by $\beta_k \in \mathbb{R}^S$ for $k = 1, 2$  
- Each individual's phenotype is then modeled as:

$$y_i = c_i^T\alpha + \sum_{s=1}^S \left(g_{i,s,1}\beta_{s,1} + g_{i,s,2}\beta_{s,2}\right) + \epsilon_i$$ 

- $c_i\in \mathbb{R}^C$ is a vector of covariates, including an intercept
- $\alpha\in \mathbb{R}^C$ is a vector of covariate effects
- $\epsilon_i$ is a random error term

## Phenotype model for admixed individuals

- Aggregating $g_{i,s,k}$ over all SNPs $s$ and all subjects $i$ gives matrices $G_{k}\in \lbrace 0,1,2\rbrace^{N\times S}$

$$y = C\alpha + G_1\beta_1 + G_2\beta_2 + \epsilon$$


- $C\in \mathbb{R}^{N \times C}$ is a matrix of covariates

## Phenotype model for admixed individuals

- Model $\beta_1, \beta_2$ as:

$$\begin{bmatrix} 
\beta_{s,1} \\
\beta_{s,2}
\end{bmatrix}
\sim N\left( 
\begin{bmatrix}
0 \\
0
\end{bmatrix}, 
\tau_s^2
\begin{bmatrix}
\frac{\sigma_g^2}{S} & \frac{\rho_g}{S} \\
\frac{\rho_g}{S} & \frac{\sigma_g^2}{S}
\end{bmatrix}
\right)
$$

- $\epsilon_i \sim N(0, \sigma_e^2)$

- $\tau_s$ denotes SNP-specific parameters for effects distribution

## Phenotype model for admixed individuals

- Define correlation of genetic effects as 

$$r_{admix} = \frac{\rho_g}{\sigma_g^2}$$

- $r_{admix} = 1$ implies that $\beta_{s,1} = \beta_{s,2}$ for all SNPs $s$
- $r_{admix} < 1$ indicates differences in causal effects between ancestries

## Specifying $\tau_s$ under different heritability models

- $\tau_s$ parameters model the coupling of SNP effects variance with MAF, local LD or other functional annotations
- Previous research has shown that genetic correlation estimation is robust to heritability model choice
- Present work's authors mainly use frequency-dependent model for both simulations and real data analyses
  - set $\tau^2_s \propto \left( f_s (1-f_s)\right)^{\alpha}$
  - $f_s$ is MAF of SNP $s$
  - $\alpha$ set to fixed value of $- 0.38$

## Evaluation of genome-wide genetic effects consistency

- Marginalize over random effects $\beta_1$ and $\beta_2$ to obtain

$$y \sim N\left(C\alpha, \sigma_g^2 \frac{G_1 T G_1^T + G_2 T G_2^T}{S} + \rho_g \frac{G_1 T G_2^T + G_2 T G_1^T}{S} + \sigma_e^2 I\right)$$

- $T$ is a diagonal matrix with $T_{ss} = \tau_s^2$ for all $s$

## Evaluation of genome-wide genetic effects consistency    

- Let $K_1 = \frac{G_1 T G_1^T + G_2 T G_2^T}{S}$ and $K_2 = \frac{G_1 T G_2^T + G_2 T G_1^T}{S}$
- Write $\rho_g = \sigma_g^2 r_{admix}$ to get 

$$y \sim N(C\alpha, \sigma_g^2 (K_1 + r_{admix} K_2) + \sigma_e^2 I)$$


## Evaluation of genome-wide genetic effects consistency

- While MLE of $(\alpha, \sigma_g^2, r_{admix}, \sigma_e^2)$ can be found by maximizing $L(\alpha, \sigma_g^2, r_{admix}, \sigma_e^2)$, the constraint that $|r_{admix}| \leq 1$ is not easy to enforce
- Authors use profile likelihood instead:

$L_p(r_{admix}) = max_{(\alpha, \sigma_g^2, \sigma_e^2)}L(\alpha, \sigma_g^2, r_{admix}, \sigma_e^2)$
- Perform grid search over $r_{admix}$ values to maxiximize $L_p(r_{admix})$
- For each candidate $r_{admix}$, 
  - compute $K_1 + r_{admix}K_2$
  - solve for $(\alpha, \sigma_g^2, \sigma_e^2)$ for a single variance component model with GCTA
  - in practice, compute $L_p(r_{admix})$ for $r_{admix} \in \lbrace 0, 0.05, 0.1, \ldots, 1\rbrace$
  - use natural cubic splines to interpolate pairs of $(r_{admix}, L_p(r_{admix}))$ to get a smooth curve
  - set $\widehat{r_{admix}}$ to value that maximizes the likelihood curve
  - set credible interval as highest posterior density interval (assuming prior $U(0,1)$ for $r_{admix}$) 


## Evaluation of genetic effects consistency at individual variant with marginal effects

- For an individual SNP $s$ and phenotype:

$$y = C\alpha + g_{s,1}\beta_{s,1}^{(m)} + g_{s,2}\beta_{s,2}^{(m)} + \epsilon$$

- Here, $\beta_{s,1}^{(m)}$ and $\beta_{s,2}^{(m)}$ are the marginal effects of the SNP 
- Marginal effects tag effects from nearby causal SNPs with taggability as a function of ancestry-specific LD with causal SNPs
- Heterogeneity in marginal effects by local ancestry can be induced even if causal effects are the same


## Evaluation of genetic effects consistency at individual variant with marginal effects

- Jointly estimate $\beta_{s,1}^{(m)}$ and $\beta_{s,2}^{(m)}$ with least squares  
- Hypothesis testing by comparing above model to:

$$y = C\alpha + (g_{s,1} + g_{s,2})\beta_{s}^{(m)} + \epsilon$$


## Marginal effects-based methods for estimating heterogeneity 

- Inputs: estimated marginal effects $\widehat{\beta_{s,1}^{(m)}}$ and $\widehat{\beta_{s,2}^{(m)}}$ and their standard errors  
- Calculations:  
  - Pearson Correlation  
  - OLS Regression Slope: $$\widehat{\beta_{s,1}^{(m)}} \sim \widehat{\beta_{s,2}^{(m)}}$$  
    - Fails to model errors in independent variable  
    - Assumes homogenous errors in dependent variable across SNPs  
  - Deming Regression Slope: $$\widehat{\beta_{s,1}^{(m)}} \sim \widehat{\beta_{s,2}^{(m)}}$$ with SEs  
    -  Deming regression models heterogeneous errors in both independent and dependent variables 
    -  More robust than above methods  

## Deming Regression

- Inputs: $y_i$ and $x_i$, $\sigma_{x, i}$ and $\sigma_{y, i}$ for $i = 1, \ldots, n$  
- Optimizes the following objective function:

$$\min_{\beta, \alpha, \delta_1, \ldots, \delta_n, \epsilon_1, \ldots, \epsilon_n} \sum_{i=1}^n \left[ \frac{\epsilon_i^2}{\sigma^2_{y,i}} + \frac{\delta_i^2}{\sigma^2_{x, j}} \right]$$

subject to: 

$$y_i + \epsilon_i = \alpha + \beta (x_i + \delta_i)$$

for $i = 1, \ldots, n$


