<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Large uncertainty in individual polygenic risk score estimation impacts PRS-based risk stratification</title>
    <meta charset="utf-8" />
    <meta name="author" content="Presented by Fred Boehm" />
    <meta name="date" content="2022-06-10" />
    <script src="zhou-lab-meeting-June-2022_files/header-attrs/header-attrs.js"></script>
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# Large uncertainty in individual polygenic risk score estimation impacts PRS-based risk stratification
]
.subtitle[
## Ding, et al. (2022)
]
.author[
### Presented by Fred Boehm
]
.date[
### June 10, 2022
]

---















class: center middle


# Prediction intervals for PGS





---

## Existing approaches to prediction intervals in PRS

1. Mondrian cross-conformal prediction intervals for PRS [Sun+21]
  - One approach to conformal prediction 
1. Bayesian credible intervals for PRS [Din+21]
  - ldpred2 used to obtain posterior samples
  - Observe large variances in PRS

---

## Importance of prediction intervals for PGS

- Clinical utility of an interval estimate in addition to point estimate
- Large variability in PRS with ldpred2 


---

class: center middle


# Methods

---

## Individual PRS uncertainty


`$$y_i = x_i^T\beta + \epsilon$$`

`$$\epsilon_i \sim N(0, \sigma^2_e)$$`

`$$GV_i = x_i^T\beta$$`

`$$\widehat{PRS}_i = x_i^T\hat \beta$$`


---

## Estimating individual uncertainty in Bayesian PRS models

- LDpred2 assumes:

`$$\beta_j \sim 
\begin{cases}
    N(0, \frac{h^2_g}{Mp_{causal}}),&amp; \text{with probability } p_{causal}\\
    0,              &amp; \text{with probability } 1 - p_{causal}
\end{cases}$$`

- `\(\hat\beta_{GWAS}\)`: marginal effects from GWAS samples
- `\(\hat R\)`: LD matrix from GWAS samples

- LDpred2 uses MCMC to sample from the posterior distribution:

`$$p(\beta | \hat\beta_{GWAS}, \hat R, h_g^2, p_{causal})$$`

---

## Estimating individual uncertainty in Bayesian PRS models

- Use posterior samples to construct a level `\(\rho\)` credible interval for `\(GV_i\)` for each subject
- For `\(B\)` posterior samples, choose the `\(\frac{1 - \rho}{2}\)` and `\(\frac{1 + \rho}{2}\)` quantiles of `\(x_i^T\tilde \beta\)` as the credible interval

---

## PRS uncertainty analytical form under infinitesimal model

- Assuming all `\(M\)` SNPs are independent &amp; causal, and effect sizes are iid draws from `\(N(0, \frac{h_g^2}{M})\)`:

`$$\hat\beta_{GWAS, j} \sim N\left(\beta_j, \frac{1}{N}\left(1 - \frac{h_g^2}{M}\right)\right)$$`
and 

`$$\mathbb{E}\left(var(x_i^T\beta | x_i, X, y, h^2_g) \right) \approx \left(\frac{1}{h_g^2} + \frac{N}{M} \right)^{-1}$$`

---

## Simulations

- Start with UKB genotypes  
- After SNP filtering, 459,792 SNPs remained  
- 291,273 white British individuals after filtering

`$$\beta_j \sim 
\begin{cases}
    N(0, \frac{h^2_g}{Mp_{causal}}),&amp; \text{with probability } p_{causal}\\
    0,              &amp; \text{with probability } 1 - p_{causal}
\end{cases}$$`

`$$(y_1, \ldots, y_N) \sim N(X\beta, (1-h_g^2)I_N)$$`

---

## 20 Parameter settings for simulations

- `\(h_g^2\)`: 0.05, 0.1, 0.25, 0.5, 0.8  
- `\(p_{causal}\)`: 0.001, 0.01, 0.1, 1  
- 10 traits per setting  
- training set: 250,000  
- validation set: 20,000  
- test set: 21,273  






---

class: center middle

# Results


---

## LD and finite GWAS sample size introduce uncertainty into PRS estimation


&lt;img src="ding_figures/41588_2021_961_Fig1_HTML.png" width="600" height="400" /&gt;

---

## GWAS sample size and causal effect size impact the relative ordering of marginal GWAS effects at tag versus true causal SNPs

&lt;img src="ding_figures/41588_2021_961_Fig7_ESM.jpg" width="600" height="400" /&gt;

???

We simulated a GWAS of N individuals ($X_{N×3}$) for 3 SNPs with LD structure R (SNP2 and SNP3 are in LD of 0.9 whereas SNP1 is uncorrelated to other SNPs) where SNP1 and SNP2 are causal with the same effect size `\(\beta_c = (\beta, \beta, 0)\)` such that the variance explained by this region is `\(var(X\beta_c) = \frac{0.5}{m_{causal}}\)`, corresponding to a trait with total heritability of 0.5 equally distributed across `\(m_{causal}\)` regions in the genome. For each parameter setting we quantified the proportion of times the marginal GWAS effect at SNP3 (tag SNP) is larger than the observed marginal effect at SNP2 (true causal) across 1,000 randomly drawn GWASs. To explore the impact of different causal effect sizes, we varied `\(m_{causal}\)` from 1,000 to 10,000 causal regions in the genome.








---

## Extracting uncertainty from Bayesian methods for probabilistic PRS-based stratification

&lt;img src="ding_figures/41588_2021_961_Fig2_HTML.png" width="600" height="400" /&gt;


---

## Expected sd is highly correlated with average sd across testing subjects

.pull-left[

&lt;img src="ding_figures/41588_2021_961_Fig3_HTML.png" width="600" height="400" /&gt;

]

.pull-right[
`$$\mathbb{E}\left(var(x_i^T\beta | D, h^2_g)\right) \approx \left(\frac{1}{h_g^2} + \frac{N}{M}\right)^{-1}$$`

]


---

## Analytical estimator of sd provides approximately unbiased estimates of average sd of testing individuals

&lt;img src="ding_figures/41588_2021_961_Fig8_ESM.jpg" width="1100" height="400" /&gt;







---

## Genetic architecture and sample size impact uncertainty in PRS estimates in simulations

.pull-left[

&lt;img src="ding_figures/41588_2021_961_Fig4_HTML.png" width="600" height="400" /&gt;
]



.pull-right[
`$$\text{Scaled sd} = \frac{\text{sd}}{\text{sd of point estimates in the testing group}}$$`
]


---

## Calibration of `\(\rho\)`-level genetic value credible interval with respect to proportion of causal effects and SNP-heritability in testing individuals

&lt;img src="ding_figures/41588_2021_961_Fig9_ESM.jpg" width="600" height="400" /&gt;


---

## Distribution of individual PRS absolute standard deviation with respect to polygenicity under different heritability

&lt;img src="ding_figures/41588_2021_961_Fig10_ESM.jpg" width="900" height="400" /&gt;

---

## Distribution of individual PRS absolute standard deviation with respect to heritability under different polygenicity

&lt;img src="ding_figures/41588_2021_961_Fig11_ESM.jpg" width="900" height="400" /&gt;





---

## Uncertainty in real data and its influence on PRS-based stratification

&lt;img src="ding_figures/41588_2021_961_Fig5_HTML.png" width="600" height="400" /&gt;


---

## Stratification uncertainty at different thresholds and credible set levels

&lt;img src="ding_figures/41588_2021_961_Fig6_HTML.png" width="600" height="400" /&gt;


---

## Individual ranking is consistent when ranking by PRS estimates versus probability of genetic value above threshold

&lt;img src="ding_figures/41588_2021_961_Fig16_ESM.jpg" width="600" height="400" /&gt;




---

## References

Ding, Y., K. Hou, K. S. Burch, et al. (2021). "Large uncertainty in
individual polygenic risk score estimation impacts PRS-based risk
stratification". In: _Nature Genetics_. ISSN: 1061-4036, 1546-1718.
DOI:
[10.1038/s41588-021-00961-5](https://doi.org/10.1038%2Fs41588-021-00961-5).
URL:
[https://www.nature.com/articles/s41588-021-00961-5](https://www.nature.com/articles/s41588-021-00961-5)
(visited on Dec. 25, 2021).

Sun, J., Y. Wang, L. Folkersen, et al. (2021). "Translating polygenic
risk scores for clinical use by estimating the confidence bounds of
risk prediction". In: _Nature Communications_ 12.1, p. 5276. ISSN:
2041-1723. DOI:
[10.1038/s41467-021-25014-7](https://doi.org/10.1038%2Fs41467-021-25014-7).
URL:
[https://www.nature.com/articles/s41467-021-25014-7](https://www.nature.com/articles/s41467-021-25014-7)
(visited on Oct. 10, 2021).


    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"slideNumberFormat": "%current%",
"highlightStyle": "github",
"highlightLines": true,
"ratio": "16:9",
"countIncrementalSlides": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
