---
title: "Adjusting for common variant polygenic scores improves yield in rare variant association analyses"
format: revealjs
date: July 19, 2023
---

## Introduction

- Many PGS use common variants
- Can including PGS in rare variant association tests improve power?

## Overview

```{r}
knitr::include_graphics(here::here("umich", "skg-lab-journal-club-july-2023", "figures", "41588_2023_1342_Fig1_HTML.png"))
```

::: {.notes}
- 65 quantitative traits  
- 3 GWAS per trait  
- 2 PGS per GWAS  
    - C + T (P < 5 × 10−8 and $r^2$ < 0.001)
    - PRS-CS  
-  exome-wide testing of rare variants within the WES samples, using models without PGSs and adjusting for various PGSs  
-  $\lambda_{GC}$, inflation factor computed as observed $\chi^2$ at the median over the expected under the null hypothesis.
:::


## Study population

- UK Biobank
  - 500,000 UK individuals aged 40–69 years at enrollment time

## Phenotypes

- 65 quantitative traits  
    - anthropometric traits  
    - metabolic blood markers  
    - blood pressure traits  
    - blood count traits  
- adjusted for lipid-lowering medication use  
- rank-based inverse normalized  

## Genetics data 

- genome-wide imputed data  
    - all common variant analyses    
    - removed imputed variants with MAF < 1% and imputation accuracy ‘INFO’ < 0.3  
- WES data  
    - RV analyses
    - 188,062 individuals (after QC)

## Common variant association analyses

- 3 GWAS per trait
    - out-of-sample GWAS: 
        - not included in the WES cohort and unrelated to the WES cohort
    - in-sample GWAS:
        - WES cohort
    - total GWAS:
        - all European individuals with imputed genotype data

## Common variant association analyses

- linear whole-genome ridge regression model
    - adjusting for sex, age, age$^2$, genotyping array and ancestral principal components 1 through 20




## PGS derivation

- 2 PGS per GWAS
- C + T ("lead SNP")
    - $r^2 < 0.001$ & $p < 5 E-8$
- PRS-CS-auto
    - 1.1 million HapMap variants only




::: {.notes}

PRS-CS-auto applies a Bayesian regression framework to identify posterior variant effect sizes based on a continuous shrinkage prior, which is directly learnt from the data

:::


## Variance explained by PGS

- phenotypic variance explained by each PGS for each trait in the null model
    - linear regression with unrelated subset of individuals with WES data
    - adjusting for sex, age, age$^2$, genotyping array and ancestral principal components 1 through 20
    - extract $R^2$ values for all models
    - variance explained by PGS for a given trait was defined as the improvement in $R^2$ in the model with PGS as compared with the model with no PGS

## Rare variant association analyses

- gene-based rare variant collapsing tests across the exome for each trait  
- grouped and analyzed LOF and predicted-deleterious missense variants per gene  
- included variants with MAC $\le$ 40, which also had MAF $<$ 0.1% in each continental population in gnomAD v.2 exomes
- GENESIS linear mixed effects models
    - adjusting for sex, age, age$^2$, genotyping array, sequencing batch, ancestral principal components 1 through 20 and a sparse kinship matrix  
    - With and without PRS as a fixed effect covariate  
- Sample sizes for the rare variant analyses ranged from n = 142,709 to n = 187,890  


## Assessment of rare variant discovery yield


- Calculate number of gene associations for each model across all traits at various significance thresholds  
- tested whether the addition of the PGS improved the significance of gene–phenotype associations  
    - two-sided paired Wilcoxon signed-rank tests to assess the improvement in −log10(P) values between two models
    - For a given comparison between two models, they included any gene–phenotype pair reaching the cutoff in either model
- To quantify the difference, $\bar{d}$, in −log10(P) values, we repeated this analysis using paired t-tests
    - For paired t-tests, we removed any gene–phenotype pair for which the difference between both models fell outside of 4 s.d. from the mean of differences.


## Figure 2


```{r}
knitr::include_graphics(here::here("umich", "skg-lab-journal-club-july-2023", "figures", "41588_2023_1342_Fig2_HTML.png"))
```

::: {.notes}
Bar charts for the improvement in deleterious RVAT yield after PGS adjustment at different $\alpha$ levels, expressed in percentage relative to the no-PGS (reference) model. 

b, Violin plots for the difference ($\delta$) in significance of tests from deleterious RVATs, comparing models with PGS versus models without PGS. Here, the $\delta$ in P values (on the −log10 scale) is displayed for tests reaching $P < 2.6 × 10−6$ (Methods). The P values and distributions are derived from two-sided paired Wilcoxon signed-rank tests (where n gene–trait pairs equals 263, 270, 258, 260, 265 and 278 from left to right), while the $\bar d$ values plotted above the violins are derived from two-sided paired t-tests (after removing outliers). The left plot shows all results, while the right plot is capped at y = 10 for clarity. 

c, Quantile–quantile plots for PGS-adjusted RVAT of the phenotype height. The left plot shows expected versus observed P values for the model with no PGS adjustment, while the second and third plots show results for PGSlead-SNP (out-sample) and PGSCS (out-sample), respectively. Exome-wide significant genes are annotated with gene names; genes highlighted in bold were only identified after PGS adjustment. $\bar d$, estimated paired group difference; $\delta$, difference; $\alpha$, significance cutoff. Ref, reference model (that is, the baseline model without inclusion of PGS).

:::




## Secondary analyses of discovery power

- assessed the power change for PGS adjustment across a range of binary traits
    - logistic mixed effects models  
- assess how PGS adjustment affects discovery power in other RVAT software  
    - collapsing tests with standard linear regression  
    - burden tests in fastGWA  
    - collapsing tests in BOLT-LMM  
    - SKAT-O tests in speed-optimized SAIGE-GENE+  
    - assessed how addition of external PGS affected power in REGENIE
- assessed a LOCO PGS and compared it with the full PGSCS for two traits
    - height & LDL cholesterol  
    - whether the power improvement was due mainly to proximal or distal common variants  


## Associations between trait heritability and PGS variance explained with yield improvement

- assessed whether the improvement in RVAT associations after PGS adjustment was associated with trait heritability or the variance explained by PGS  
    - estimate $h^2_{SNP}$ for each trait  
    - using the total sample GWAS  
    - using the baselineLD_2.2 file from the LDSC software as the linkage disequilibrium reference  
- linear regression to regress the change in number of trait RVAT associations on the estimated $h^2_{SNP}$
- linear regression to regress the change in number of trait RVAT associations on the R2 of the PGS for its respective traits


## Assessment of false-positive rate using rare synonymous variation

- Synonymous variants are generally not expected to affect the amino acid sequence encoded by genes, and therefore are strongly depleted of true genetic effects  
- grouped rare synonymous variants (MAC ≤ 40 and MAF < 0.1% in each continental population in gnomAD exomes) and ran exome-wide gene-based collapsing tests using GENESIS  


## Assessment of exome-wide inflation

- Exome-wide test statistics were plotted in quantile–quantile (QQ) plots to visually assess inflation per trait, per model, per variant mask  
- Exome-wide inflation was further quantified using $\lambda_{GC}$ values  
    - $\lambda_{GC}$ was computed as the median of the observed $\chi^2$ values divided by the median of the expected $\chi^2$ values under the null hypothesis  




## Figure 3

```{r}
knitr::include_graphics(here::here("umich", "skg-lab-journal-club-july-2023", "figures", "41588_2023_1342_Fig3_HTML.png"))
```


::: {.notes}

a, Boxplots for per-trait association rate from synonymous RVATs at different α levels across the 65 traits. A median of 18,060 genes were analyzed per trait. 

b, Violin plots for genomic inflation factors for exome-wide RVATs of synonymous variants across the 65 traits. 

c, Violin plots for difference ($\delta$) in significance of tests from synonymous variant RVATs, comparing models with PGSs versus models without PGSs. Here, the $\delta$ in P values (on the −log10 scale) is displayed for tests reaching P < 0.05 (Methods), with the contributing n gene–trait pairs equaling 75,044, 77,524, 75,838, 89,792, 77,784 and 85,187 (from left to right). Boxplots: center line, median; box limits, upper and lower quartiles; whiskers, 1.5 × interquartile range; points, outliers

:::




