---
title: "Enhancing QTL mapping with allelic series inferences"
author: "Fred Boehm"
institute: "U. Mass. Medical School"
date: "October 14, 2020 (updated: `r Sys.Date()`)"
bibliography: ../research.bib
output:
  xaringan::moon_reader:
    css: xaringan-themer.css
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true  
      countIncrementalSlides: false
---

```{r setup2, include=FALSE, dev="CairoPNG"}
knitr::opts_chunk$set(dev="CairoPNG")
```

```{r xaringan-themer, include = FALSE}
library(xaringanthemer)
style_mono_light(
  base_color = "#1c5253",
  header_font_google = google_font("Josefin Sans"),
  text_font_google   = google_font("Montserrat", "300", "300i"),
  code_font_google   = google_font("Droid Mono")
)
```

```{r, echo = FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
```

```{r}
PATH_TO_DDATA <- "~/work/research/qtl2tb/analysis/data/derived_data"
#PATH_TO_DDATA <- "~/qtl2tb/analysis/data/derived_data" #for use on ghpcc
library(magrittr)
nc <- 16
```

# DO QTL mapping study

- 763 Diversity Outbred mice
- GigaMUGA genotyping array
- Phenotypes: Lung CFU, Spleen CFU, peak weight, weight at death, percent normal lung
- Covariates: Sex, Batch, Dose



---

# QTL mapping results in DO

```{r}
load(file.path(PATH_TO_DDATA, "fda_qtl_input.Rdata"))
fn <- file.path(PATH_TO_DDATA, "scan1out_do-clinical.rds")
qtl <- readRDS(fn)
qtl <- do.call("cbind", qtl)
ls_ratio <- qtl2::find_peaks(qtl[, ncol(qtl), drop = FALSE], map, threshold = 6, peakdrop = 1.5, prob = 0.95) %>% # get only one trait's peaks
  tibble::as_tibble() %>%
  dplyr::mutate(trait = "lung_spleen_ratio")
peaks <- qtl2::find_peaks(qtl[ , - ncol(qtl)], map, threshold = 6, peakdrop = 1.5, prob = 0.95) %>%# get peaks for other traits
  tibble::as_tibble() %>%
  dplyr::mutate(trait = lodcolumn) %>%
  dplyr::bind_rows(ls_ratio) %>%
  dplyr::select(trait, chr, pos, lod, ci_lo, ci_hi, - lodcolumn)
peaks %>%
    gt::gt() %>%
    gt::fmt_number(
    columns = dplyr::vars(lod, pos, ci_lo, ci_hi),
    decimals = 1
  )
```

---

# Lung CFU on Chr 17

```{r, eval = FALSE}
afn <- file.path(PATH_TO_DDATA, "do_scan1blup_lung_Chr17.rds")
if (!file.exists(afn)){
  s1b <- qtl2::scan1blup(probs[, 17], 
                       pheno_rz[ , 1, drop = FALSE], 
                       kinship = K$`17`, 
                       addcovar = covar, 
                       cores = nc)
  saveRDS(s1b, afn)
} else {s1b <- readRDS(afn)}
```

```{r}
s1c <- qtl2::scan1coef(probs[, 17], 
                       pheno_rz[ , 1, drop = FALSE], 
                       kinship = K$`17`, 
                       addcovar = covar, 
                       cores = 1)
```


```{r}
left <- names(map[[17]])[min(which(map[[17]] > 30))]
right <- names(map[[17]])[min(which(map[[17]] > 50))] 
s1 <- qtl[which(rownames(qtl) == left):which(rownames(qtl) == right), 1, drop = FALSE]
b1 <- s1c[which(rownames(s1c) == left):which(rownames(s1c) == right), ]

qtl2::plot_coefCC(b1, map, scan1_output = s1)
```


---

# Spleen CFU on Chr 17

```{r, eval = FALSE}
bfn <- file.path(PATH_TO_DDATA, "do_scan1blup_spleen_Chr17.rds")
if (!file.exists(bfn)){
  s1b <- qtl2::scan1blup(probs[, 17], 
                       pheno_rz[ , 2, drop = FALSE], 
                       kinship = K$`17`, 
                       addcovar = covar, 
                       cores = nc)
  saveRDS(s1b, bfn)
} else {s1b <- readRDS(bfn)}
```

```{r}
s1c <- qtl2::scan1coef(probs[, 17], 
                       pheno_rz[ , 2, drop = FALSE], 
                       kinship = K$`17`, 
                       addcovar = covar, 
                       cores = 1)
```

```{r}
s1 <- qtl[which(rownames(qtl) == left):which(rownames(qtl) == right), 2, drop = FALSE]
b1 <- s1c[which(rownames(s1c) == left):which(rownames(s1c) == right), ]

qtl2::plot_coefCC(b1, map, scan1_output = s1)
```

---

# Spleen CFU on Chr 1

```{r, eval = FALSE}
cfn <- file.path(PATH_TO_DDATA, "do_scan1blup_spleen_Chr1.rds")
if (!file.exists(cfn)){
  s1b <- qtl2::scan1blup(probs[, 1], 
                       pheno_rz[ , 2, drop = FALSE], 
                       kinship = K$`1`, 
                       addcovar = covar, 
                       cores = nc)
  saveRDS(s1b, cfn)
} else {s1b <- readRDS(cfn)}
```

```{r}
s1c <- qtl2::scan1coef(probs[, 1], 
                       pheno_rz[ , 2, drop = FALSE], 
                       kinship = K$`1`, 
                       addcovar = covar, 
                       cores = nc)
```  

```{r}
left <- names(map[[1]])[min(which(map[[1]] > 120))]
right <- names(map[[1]])[min(which(map[[1]] > 140))] 
s1 <- qtl[which(rownames(qtl) == left):which(rownames(qtl) == right), 2, drop = FALSE]
b1 <- s1c[which(rownames(s1c) == left):which(rownames(s1c) == right), ]

qtl2::plot_coefCC(b1, map, scan1_output = s1)
```



---

# Spleen CFU on Chr 11

```{r, eval = FALSE}
bfn <- file.path(PATH_TO_DDATA, "do_scan1blup_spleen_Chr11.rds")
if (!file.exists(bfn)){
  s1b <- qtl2::scan1blup(probs[, 11], 
                       pheno_rz[ , 2, drop = FALSE], 
                       kinship = K$`11`, 
                       addcovar = covar, 
                       cores = nc)
  saveRDS(s1b, bfn)
} else {s1b <- readRDS(bfn)}
```

```{r, eval = TRUE}
s1c <- qtl2::scan1coef(probs[, 11], 
                       pheno_rz[ , 2, drop = FALSE], 
                       kinship = K$`11`, 
                       addcovar = covar, 
                       cores = 1)
  
```

```{r}
left <- names(map[[11]])[min(which(map[[11]] > 112))]
right <- names(map[[11]])[length(map[[11]])] 
s1 <- qtl[which(rownames(qtl) == left):which(rownames(qtl) == right), 2, drop = FALSE]
b1 <- s1c[which(rownames(s1c) == left):which(rownames(s1c) == right), ]

qtl2::plot_coefCC(b1, map, scan1_output = s1)
```


---

# Spleen CFU on Chr 15

```{r, eval = FALSE}
bfn <- file.path(PATH_TO_DDATA, "do_scan1blup_spleen_Chr15.rds")
if (!file.exists(bfn)){
  s1b <- qtl2::scan1blup(probs[, 15], 
                       pheno_rz[ , 2, drop = FALSE], 
                       kinship = K$`15`, 
                       addcovar = covar, 
                       cores = nc)
  saveRDS(s1b, bfn)
} else {s1b <- readRDS(bfn)}
```

```{r}
s1c <- qtl2::scan1coef(probs[, 15], 
                       pheno_rz[ , 2, drop = FALSE], 
                       kinship = K$`15`, 
                       addcovar = covar, 
                       cores = 1)
```

```{r}
left <- names(map[[15]])[min(which(map[[15]] > 70))]
right <- names(map[[15]])[min(which(map[[15]] > 85))] 
s1 <- qtl[which(rownames(qtl) == left):which(rownames(qtl) == right), 2, drop = FALSE]
b1 <- s1c[which(rownames(s1c) == left):which(rownames(s1c) == right), ]

qtl2::plot_coefCC(b1, map, scan1_output = s1)
```



---

# Reduced models in CC mice

---

# CC clinical QTL

```{r, scan1-cc}
load(file.path(PATH_TO_DDATA, "reduced_map_qtl2_mapping_objects.Rdata"))
cc_kinship <- qtl2::calc_kinship(genoprobs.1, "loco")
cc_pheno <- read.csv(file.path(PATH_TO_DDATA, "batch_corrected_phenos_summarized_05OCT2019_v4.csv"))[, -c(1:7)]
rownames(cc_pheno) <- read.csv(file.path(PATH_TO_DDATA, "batch_corrected_phenos_summarized_05OCT2019_v4.csv"))[, 1] %>% paste0("cs")

cc_s1 <- qtl2::scan1(genoprobs = genoprobs.1, 
                     pheno = cc_pheno, 
                     kinship = cc_kinship, 
                     addcovar = addcovar, 
                     reml = TRUE
                     )
```


```{r tnseq-timbr}
t_out <- readRDS(file.path(PATH_TO_DDATA, "cc-timbr-clinical-M1-with-tnseq-reduced-peaks-cis-perms-pvalues.rds")) 
t_out %>%
  dplyr::select(lodcolumn, chr, pos, lod, cc_peak_bayes_lower, cc_peak_bayes_upper) %>%
  dplyr::rename(trait = lodcolumn, lower = cc_peak_bayes_lower, upper = cc_peak_bayes_upper) %>%
  gt::gt() %>%
    gt::fmt_number(
    columns = dplyr::vars(lod, pos, lower, upper),
    decimals = 1
  )
```

---

# Reduced models

### Full model

$$\text{trait} = p_A (\text{effect}_A) + ... + p_H (\text{effect}_H) + \text{error}$$

### Reduced model

$$\text{trait} = p_1(\text{effect}_1) + p_2(\text{effect}_2) + \text{error}$$

where, perhaps, $p_1 = p_A + p_B + p_C + p_D + p_E$ and $p_2 = p_F + p_G + p_H$
with 

$\text{effect}_A = ... = \text{effect}_E$ and $\text{effect}_F = \text{effect}_G = \text{effect}_H$

---

# TIMBR for inferring allelic series

- Use peak marker
- Samples from posterior distribution 
  - Each sample has an allelic series (with up to 8 alleles)
  - Each sample has effects for its allelic series
- I choose the most frequently occurring allelic series when using reduced models

---

# TIMBR results for lung CFU on Chr 7

```{r}
t_out$timbr...25[[1]]$post.hap.effects %>%
  qtl2tb::kde_plot()
```
---

# Chr 7 lung CFU

```{r, gp}
qtl2tb::gp_scatter(geno = genoprobs.1[[7]][ , , t_out$marker_index[1]], 
                   pheno = cc_pheno[stringr::str_starts(rownames(cc_pheno), "CC"), 1, drop = TRUE]) %>%
plotly::ggplotly()
```

---

# Allele effects at Chr 7 lung CFU CC QTL

```{r}
left <- names(map.1[[7]])[1]
right <- names(map.1[[7]])[min(which(map.1[[7]] > 25))] 
s1b_lungcfu_chr7 <- qtl2::scan1blup(genoprobs = genoprobs.1[, 7], 
                pheno = cc_pheno[, 1, drop = FALSE], 
                kinship = cc_kinship[[7]], 
                addcovar = addcovar, 
                se = TRUE
                )
s1 <- cc_s1[which(rownames(cc_s1) == left):which(rownames(cc_s1) == right), 1, drop = FALSE]
b1 <- s1b_lungcfu_chr7[which(rownames(s1b_lungcfu_chr7) == left):which(rownames(s1b_lungcfu_chr7) == right), ]
qtl2::plot_coefCC(b1, map = map.1, scan1_output = s1)
```

---

# Reduced scans of Tnseq traits

1. Infer allelic series at a CC clinical trait QTL  
1. Reduced scans of CC Tnseq traits over Bayes credible interval of the CC clinical QTL
1. Identify peaks for Tnseq traits
1. Permutation test to get p-values
1. (Not yet implemented) Multiple trait testing adjustment

---

# Reduced scans of Tnseq traits (lung CFU, Chr 7)

```{r}
t_out$cis_peaks[[1]] %>%
  dplyr::select(-c(ci_lo, ci_hi, lodindex)) %>%
  dplyr::rename(trait = lodcolumn) %>%
  dplyr::arrange(pvalue) %>%
  dplyr::filter(!stringr::str_starts(trait, "rv")) %>%
  dplyr::filter(pvalue < 0.05) %>%
  gt::gt() %>%
    gt::fmt_number(
    columns = dplyr::vars(lod, pos),
    decimals = 1
  )
```

How to summarize results?

How to reach biological conclusions?

???

Each CC clinical QTL has a table like this

Showing only annotated genes

---

# TIMBR results for lung CFU on Chr 15

```{r}
t_out$timbr...25[[2]]$post.hap.effects %>%
  qtl2tb::kde_plot()
```
---

# Chr 15 lung CFU

```{r, gp2}
qtl2tb::gp_scatter(geno = genoprobs.1[[15]][ , , t_out$marker_index[2]], 
                   pheno = cc_pheno[stringr::str_starts(rownames(cc_pheno), "CC"), 1, drop = TRUE]) %>%
plotly::ggplotly()
```

---

# Allele effects at Chr 15 lung CFU CC QTL

```{r}
left <- names(map.1[[15]])[min(which(map.1[[15]] > 20))]
right <- names(map.1[[15]])[min(which(map.1[[15]] > 40))] 
s1b_lungcfu_chr15 <- qtl2::scan1blup(genoprobs = genoprobs.1[, 15], 
                pheno = cc_pheno[, 1, drop = FALSE], 
                kinship = cc_kinship[[15]], 
                addcovar = addcovar, 
                se = TRUE
                )
s1 <- cc_s1[which(rownames(cc_s1) == left):which(rownames(cc_s1) == right), 1, drop = FALSE]
b1 <- s1b_lungcfu_chr15[which(rownames(s1b_lungcfu_chr15) == left):which(rownames(s1b_lungcfu_chr15) == right), ]
qtl2::plot_coefCC(b1, map = map.1, scan1_output = s1)
```


---

# Reduced scans of Tnseq traits (lung CFU, Chr 15)



```{r}
t_out$cis_peaks[[2]] %>%
  dplyr::select(-c(ci_lo, ci_hi, lodindex)) %>%
  dplyr::rename(trait = lodcolumn) %>%
  dplyr::arrange(pvalue) %>%
  dplyr::filter(!stringr::str_starts(trait, "rv")) %>%
  dplyr::filter(pvalue < 0.05) %>%
  gt::gt() %>%
    gt::fmt_number(
    columns = dplyr::vars(lod, pos),
    decimals = 1
  )
```


---

# Next steps

1. Drill down into CC Tnseq peaks near CC clinical trait peaks  
1. Incorporate DO clinical trait results
1. Process DO RNASeq data into mappable traits




---
class: inverse, center, middle

# References

