---
title: "Prediction intervals and polygenic scores"
subtitle: ""  
author: 
  - "Fred Boehm"
date: 'March 11, 2022'
bibliography: polygenic.bib
output:
  xaringan::moon_reader:
    css: xaringan-themer.css
    nature:
      slideNumberFormat: "%current%"
      highlightStyle: github
      highlightLines: true
      ratio: 16:9
      countIncrementalSlides: true
---

```{r setup, include=FALSE}
options(htmltools.dir.version = FALSE)
knitr::opts_chunk$set(
  fig.width=9, fig.height=3.5, fig.retina=3,
  out.width = "100%",
  cache = FALSE,
  echo = TRUE,
  message = FALSE, 
  warning = FALSE,
  hiline = TRUE
)
```

```{r xaringan-themer, include=FALSE, warning=FALSE}
library(xaringanthemer)
style_duo_accent(
  primary_color = "#1381B0",
  secondary_color = "#FF961C",
  inverse_header_color = "#FFFFFF"
)
```

## Outline

1. Polygenic scores 
1. Existing methods for prediction intervals for PGS 
1. DBSLMM 
1. Calculating $Var(\hat y)$ 
1. C++ Implementation 




---

class: center middle


# Polygenic scores

---

## Polygenic scores




---

## Popular PGS methods

1. DBSLMM 


---

class: center middle


# DBSLMM Model & Methods


---

## DBSLMM Model

$$y = X\beta + \epsilon$$

- trait $y$, n vector
- $X$ n by m matrix of standardized SNP genotypes 
- $\beta$ m vector of SNP effects 
- $\epsilon$ n vector of random errors with precision $\tau$


---

## DBSLMM Model

$$y = X\beta + \epsilon$$

$$\beta_j \sim \pi N(0, \sigma^2_l\tau^{-1}) + (1 - \pi) N(0, \sigma^2_s \tau^{-1})$$
- $\pi$ proportion of SNPs in the large variance component

---

## DBSLMM Model Fitting

$$y = X\beta + \epsilon$$
- BSLMM: MCMC for model fitting is slow with large memory requirements 
- Large effect SNPs should be easy to identify from GWAS analysis 
- Small effect SNPs can't be inferred accurately 
- But polygenic effects may be inferred with accuracy 

---

## DBSLMM Model

$$y = X_l\beta_l + X_s \beta_s + \epsilon$$

- $X_l$: n by $m_l$ SNP genotypes matrix for large effect SNPs 
- $\beta_l$: $m_l$ effects vector for large effect SNPs 
- $X_s$: n by $m_s$ SNP genotypes matrix for small effect SNPs 
- $\beta_s$: $m_s$ effects vector for small effect SNPs 

$$\beta_{lj} \sim N(0, \sigma_l^2\tau^{-1})$$

$$\beta_{sj} \sim N(0, \sigma_s^2\tau^{-1})$$
- Set $\sigma^2_l \to \infty$ & treat $\beta_l$ as fixed effects 


---

## DBSLMM Model: Parameter estimation

- Clumping and Thresholding (C + T) procedure in PLINK to select large effect SNPs
  - One chromosome at a time
  - p-value threshold: $10^{-6}$ 
  - region size: 1 MB 
  - LD threshold: $r^2 = 0.1$ 
- Combine large effect SNPs across genome to get $m_l$ SNPs

---

## DBSLMM Model: Parameter estimation

$$\hat \beta_l = (X_l^TH^{-1}X_l)^{-1}X_l^TH^{-1}y$$

$$\hat \beta_s = \hat\sigma^2_s X_s^TH^{-1}( y - X_l\hat\beta_l)$$

$$Var(y) = H = \hat\sigma^2_s X_sX_s^T + I_n$$

---

## DBSLMM Model: Parameter estimation

- Set $\hat\sigma^2_s$ to predetermined value instead of estimating it  
  - LD score regression to get SNP heritability, $\hat h^2$
  - Set $\hat \sigma^2_s = \frac{\hat h^2}{m}$
  
  
---

## DBSLMM Model: Parameter estimation

- Use Woodbury matrix identity to calculate $H^{-1}$

$$H^{-1} = I_n - X_s(\sigma_s^{-2}I_{m_s} + X_s^TX_s)^{-1}X_s^T$$

---

## DBSLMM: Using GWAS summary statistics 

$$z_j = \frac{X_j^Ty}{\sqrt n}$$
- $X_j$ is the $j^{th}$ column of $X$, *i.e.*, the $n$ genotypes for the $j^{th}$ SNP 

- $\Sigma_{ss} = \frac{X_s^T X_s}{n}$ 
- $\Sigma_{ll} = \frac{X_l^T X_l}{n}$ 
- $\Sigma_{sl} = \frac{X_s^T X_l}{n}$

$$\hat \beta_l = \frac{1}{\sqrt n}\left( \Sigma_{ll} - \Sigma_{ls}(\hat \sigma_s^{-2}n^{-1}I_{m_s} + \Sigma_{ss})^{-1}\Sigma_{sl}\right)^{-1}\left( z_l - \Sigma_{ls}(\hat \sigma_s^{-2}n^{-1}I_{m_s} + \Sigma_{ss})^{-1}z_s\right)$$

$$\hat \beta_s = \hat\sigma_s^2\left( I_{m_s} - \Sigma_{ss}(\hat\sigma_s^{-2}n^{-1}I_{m_s} + \Sigma_ss)^{-1}\right)(\sqrt n z_s - n \Sigma_{sl}\hat \beta_l)$$

---

## DBSLMM Model: SNP correlation matrices

- Use a small set of individuals that differ from the test and training sets 
- Approximate the SNP correlation matrix with a block diagonal matrix
- Blocks defined by LD patterns
- Block diagonal SNP correlation matrix enables blockwise calculations of $\hat\beta_l$ and $\hat\beta_s$



---

class: center middle


# Calculating $Var(\hat{\tilde y})$


---

## Calculating $Var(\hat{\tilde y})$

$$
\begin{aligned}
Var(\hat{\tilde y}) &= Var(\tilde X_l\hat\beta_l + \tilde X_s \hat\beta_s)\\
&= Var(\tilde X_l\hat\beta_l)  + Var(\tilde X_s \hat\beta_s) \\
&= \tilde X_l Var(\hat\beta_l) \tilde X_l^T + \tilde X_s Var(\hat\beta_s) \tilde X_s^T
\end{aligned}
$$

---

# C++ implementation

- Blockwise calculations of $Var(\hat{\tilde y})$
  - Each block produces a n by n matrix 
  - Sum n by n matrices to get $Var(\hat{\tilde y})$






---

## References


